### PR TITLE
Hooks Package Versions

### PR DESCRIPTION
**Pull request by marcoAntonioNina**
*Changes*

- Add Component Menu for Screen and Script Builder
- add hooks for packages
- add way for register js for Script builder

required for Package versions

### CODE REVIEW

File: resources/js/processes/screen-builder/screen.vue [hunk 1]
```diff
@@ -296,8 +361,41 @@ import formTypes from "./formTypes";
       this.updatePreview(new Object());
       this.previewInput = "{}";
       ProcessMaker.EventBus.$emit("screen-builder-start", this);
+
+      //actions buttons
+      ProcessMaker.EventBus.$on("change_mode", (value) => {
```
Comment (danloa) [inline resources/js/processes/screen-builder/screen.vue]: Uniform de use of quotes for strings, use single quotes '

File: resources/js/processes/screen-builder/screen.vue [hunk 2]
```diff
@@ -191,6 +158,102 @@ import formTypes from "./formTypes";
         items: []
       }];
 
+      const options = [
```
Comment (danloa) [inline resources/js/processes/screen-builder/screen.vue]: Uniform de use of quotes for strings, use single quotes '

File: resources/js/components/Menu.vue [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+            </b-col>
+
+        </b-row>
+    </b-card-header>
+</template>
+
+
+<script>
+  export default {
+    props: ["options"],
```
Comment (danloa) [inline resources/js/components/Menu.vue]: Uniform de use of quotes for strings, use single quotes '

File: resources/js/components/Menu.vue [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+                </template>
+
+            </b-col>
+
+            <b-col class="text-right" v-if="displayRight">
+                <template v-for="(item, index) in section.right">
+                    <b-button-group v-if="isVisible(item, 'group')" size="sm" :key="index">
+                        <b-button v-for="(button, indexButton) in item.items" :variant="button.variant || 'secondary'"
+                                  class="text-capitalize" :title="button.title" :key="indexButton"
```
Comment (caleeli) [inline resources/js/components/Menu.vue]: A better `key` could be 
```
:key="`group-${index}`"
```

**Pull request merged**